Newsletters: add 150 (2021-05-26)

harding opened the pull request:
Happy 150! :rocket: 

- [x] Add topic links @harding
- [x] Update releases/RCs @harding 
- [x] StackExchange section @bitschmidty 
- [x] `Bitcoin Core #21843` rpc: enable GetAddr, GetAddresses, and getnodeaddresses by network @jnewbery 
- [x] `Eclair #1810` Make payment_secret mandatory @dongcarl

jonatack commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,220 @@
+
+- [Eclair #1774][] extends Java's built-in `SecureRandom()` [CSPRNG][]
+  function with a secondary source of weaker randomness.  The weaker
+  randomness is hashed and the hash digest xored with the primary
```
(maybe) s/xored/XORed/

harding replied:
I don't think we've historically uppercased other boolean terms, so we probably shouldn't start now.  If you're really worried about this, I can rephrase to use "xor" without a suffix.

harding replied:
(FWIW, I just searched Google and DuckDuckGo for "xored" and got reasonable results, so anyone who does get confused should be able to straighten themselves out pretty quickly.)

jonatack replied:
not worried about it, so long as no one thinks we meant to write "bored" instead :stuck_out_tongue:

jonatack commented on _posts/en/newsletters/2021-05-26-newsletter.md:
s/wallets/wallets./

jonatack reviewed (commented):
Looking good so far on first read! (did not yet check links)

jnewbery commented on _posts/en/newsletters/2021-05-26-newsletter.md:
perhaps s/wallets/signers/ :

```suggestion
  how signers, particularly hardware signers, can securely exchange the
```

harding commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -106,7 +106,46 @@ answers posted since our last update.*
+  Murch calculates potential WU for P2TR coinjoin transactions for participants
+  utilizing two input, two output payments using cross input signature aggregation as:
+
+  - 1 participant coinjoin per participant cost: 783 WU
```
Maybe call this "Base cost for two inputs and two outputs" since a "1 participant coinjoin" doesn't make sense by itself.

harding replied:
Also, our style guide says "use vbytes rather than weight units, unless using weight units would significantly improve conciseness or comprehensibility."  You can just divide all of @Xekyo's figures by 4 to get vbytes (up to you whether or not to include any decimal remainders; I usually include them).

If you really want to use weight units (which is fine by me), I suggest at least slWU/Weight Units (WU)/ for the first occurrence of the term.

jnewbery replied:
This question was removed. Hopefully we can include it in next month's SE section.

jnewbery reviewed (commented):
One suggested change to the Stack Exchange section. Otherwise looks great.

bitschmidty commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -159,14 +159,21 @@ repo], [Hardware Wallet Interface (HWI)][hwi repo],
 
+- [Eclair #1810][] makes it mandatory for peers to signal and comply with the
+  `payment_secret` feature bit. The `payment_secret` feature thwarts the
+  de-anonymization attack whereby intermediary peers on a payment path can
```
Do we have a good Optech reference to provide for this attack? @dongcarl

dongcarl replied:
Hmm... Looks like we have:
https://bitcoinops.org/en/newsletters/2019/12/04/#c-lightning-3259
https://bitcoinops.org/en/newsletters/2020/11/04/#c-lightning-4162

dongcarl replied:
Will update, sec

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,266 @@
+mission that we continue to believe in, and that guides all the work we do.
+Since June 2018, we've:
+
+* Published 150 [newsletters][], numerous [blog posts][] and field reports,
+  a [special series on bech32][bech32], and an [interactive taproot workshop][]. In
```
Maybe:
```suggestion
* Published 150 [newsletters][], numerous [blog posts][], field reports,
  a [special series on bech32][bech32], and an [interactive taproot workshop][]. In
```

jnewbery replied:
The 'numerous' refers to both 'blog posts' and 'field reports', so keeping this as one list item.

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,266 @@
+technical work happening in the open source Bitcoin community. Although we
+couldn't foresee exactly what form that would take three years ago, it's a
+mission that we continue to believe in, and that guides all the work we do.
+Since June 2018, we've:
```
Maybe:

```suggestion
Since June 2018, we have:
```

jnewbery replied:
I think either "we have" or "we've" are fine.

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,266 @@
+This week's newsletter announces a change of networks for several IRC
+channels and celebrates Optech's 150th newsletter.  Also included are
+our regular sections with popular questions and answers from the Bitcoin
+StackExchange, new software releases and release candidates, and notable
```
```suggestion
Stack Exchange, new software releases and release candidates, and notable
```

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,266 @@
+want to contribute, or have suggestions for how we can improve, please don't
+hesitate to contact us at [info@bitcoinops.org][info].
+
+## Selected Q&A from Bitcoin StackExchange
```
```suggestion
## Selected Q&A from Bitcoin Stack Exchange
```

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,266 @@
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
```
```suggestion
*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
```

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```diff
@@ -0,0 +1,266 @@
+  `getblock`'s versionhex field, where the signaling version bits reside, to
+  observe signaling.
+
+- [What block space savings would we get for coinjoins (and payjoins) if we had cross input signature aggregation?]({{bse}}106224)
```
I was planning on writing a better answer for that one anyway, I can make sure that it happens today.

I would agree, though, that we might want to drop it otherwise.

jnewbery replied:
I look forward to reading it :)

We can either include it here or in next month's SE section. It's not going to go out of date.

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```suggestion
  the `m/48'` standard) or create new standards for things that can be
```

Xekyo commented on _posts/en/newsletters/2021-05-26-newsletter.md:
```suggestion
  key derivation paths a wallet should support.  Path templates provide
  a compact way for the user to specify which paths they
  want to use. The compactness of path templates makes it easy to back
```

Xekyo reviewed (commented):
I noticed that we sometimes write "Bitcoin StackExchange" and sometimes "Bitcoin Stack Exchange". FWIW, the "Stack Exchange" network writes itself with a space in the sites' title bars, even though the logo at best has a thin space. I don't care much either way, but we should be consistent.

adamjonas commented:
Didn't see anything else that wasn't already noted. 

Thanks for the 150th edition write-up @jnewbery. Heck of a track record y'all are putting together.

harding commented:
Addressed all feedback for my sections, plus some other feedback that seemed important to me.  There are a few nitpicks for @jnewbery left that (IMO) can be either taken or left with no issues.  Only one topic link to add this week.

Xekyo commented:
LGTM

jnewbery commented:
ACK 9a3e0fc7d6d3c6544f2bbf015f71a05dbc3980f7

All review comments have been addressed. Topic links look good. Thanks everyone!
